### PR TITLE
Adds mirror/urinal deconstruction

### PR DESCRIPTION
**Pull request by Goldsteel**
[featurerequest][qol][sprites]

Cockswain requested this. You can now remove these items when remodelling the toilets as a construction zone.

image

image

:cl:
 * rscadd: You can now deconstruct mirrors and urinals with a crowbar.
 * tweak: Urinals can be unanchored with a wrench now.
 * imageadd: Urinals have a rear sprite.

### CODE REVIEW

**Comment by Goldsteel** [comment]
Despite using getFromPool(/obj/item/stack/sheet/metal, loc, 1) I think this is done, it just depends on if there's any strange interaction with this and anything else

**Comment by brndd** [comment]
Urinals probably need a sideways sprite too. From what I remember the existing one is absolutely atrocious.

File: code/game/objects/structures/mirror.dm [hunk 1]
```diff
@@ -107,6 +107,19 @@
 			icon_state = "mirror"
 			playsound(src, 'sound/items/Deconstruct.ogg', 80, 1)
 
+	else if(istype(I, /obj/item/weapon/crowbar))
+		to_chat(user, "<span class='info'>You begin to disassemble the mirror.</span>")
```
Comment (Kurfursten) [inline code/game/objects/structures/mirror.dm]: use notice class for this

File: code/game/objects/structures/watercloset.dm [hunk 2]
```diff
@@ -183,6 +183,20 @@
...
 
 /obj/structure/urinal/attackby(obj/item/I as obj, mob/user as mob)
+	if(iswrench(I))
+		to_chat(user, "<span class='notice'>You [anchored ? "un":""]bolt \the [src]'s grounding lines.</span>")
+		anchored = !anchored
+	if(anchored == 0)
+		return
+
+	if(istype(I, /obj/item/weapon/crowbar))
+		to_chat(user, "<span class='info'>You begin to disassemble the urinal.</span>")
```
Comment (Kurfursten) [inline code/game/objects/structures/watercloset.dm]: same

File: code/game/objects/structures/watercloset.dm [hunk 3]
```diff
@@ -183,6 +183,20 @@
 	return ..()
 
 /obj/structure/urinal/attackby(obj/item/I as obj, mob/user as mob)
+	if(iswrench(I))
+		to_chat(user, "<span class='notice'>You [anchored ? "un":""]bolt \the [src]'s grounding lines.</span>")
+		anchored = !anchored
+	if(anchored == 0)
```
Comment (Kurfursten) [inline code/game/objects/structures/watercloset.dm]: it's not a big deal but I think this is more elegantly expressed as !anchored

File: code/game/objects/structures/watercloset.dm (hunk 2 above)
Comment (Kurfursten) [reply code/game/objects/structures/watercloset.dm]: also better as \the [src]

**Comment by Kurfursten** [comment]
It's not really a big deal if someone deconstructs all the urinals but it's kind of annoying if someone deconstructs all the mirrors, those are used for hair changing and shit, players need a way to get more/build more.

**Comment by Goldsteel** [comment]
We have pocket mirrors from cargo iirc

File: code/game/objects/structures/watercloset.dm (hunk 3 above)
Comment (Goldsteel) [reply code/game/objects/structures/watercloset.dm]: done

File: code/game/objects/structures/watercloset.dm (hunk 2 above)
Comment (Goldsteel) [reply code/game/objects/structures/watercloset.dm]: done

File: code/game/objects/structures/mirror.dm (hunk 1 above)
Comment (Goldsteel) [reply code/game/objects/structures/mirror.dm]: done

**Pull request merged**